ADR/adr-0008-secret-mgmt-for-user-workloads.md

```diff
@@ -0,0 +1,101 @@
+* As a StoneSoup component, I expect to have the ability to securely upload a secret, associate it with the component, and propagate it to the target environment.
+
+## Decision
+
+### Architecture Overview
+#### Terminology
+
+- `UploadSecret`: A short-lived Kubernetes secret used to deliver confidential data to permanent storage and link it to the RemoteSecret CR.
+- `SecretData`: An object stored in permanent SecretStorage. Valid SecretData is always linked to a RemoteSecret CR.
+- `RemoteSecret`: A CRD that appears during upload and links `SecretData` + `DeploymentTarget(s)` + K8s `Secret`. `RemoteSecret` is linked to one (or zero) SecretData and manages its deleting/updating.
+- K8s `Secret`: What appears at the output and is used by consumers.
+- `SecretId`: A unique identifier of SecretData in permanent SecretStorage.
+- `SecretStorage`: A database eligible for storing `SecretData` (such as HashiCorp Vault, AWS Secret Manager).
+
+The idea is to introduce a new CR called `RemoteSecret`, which is a Kubernetes representation of the `K8s Secret` that is stored in permanent storage aka SecretStorage. This CR contains a reference to the destination, either a Kubernetes namespace or an `Environment`. `UploadSecret` is used to perform an upload to the permanent storage and is represented as a regular Kubernetes secret with special labels and annotations recognized by the _SPI controller_. Different implementations of `SecretStorage` can be used, such as AWS Secret Manager or HashiCorp Vault.
+
+#### Example: If the destination is `Environment`
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: jdbc-connection-parameter
+  labels:
+    spi.appstudio.redhat.com/upload-secret: secret
+    spi.appstudio.redhat.com/target-environment: prod
+type: Opaque
+data:
+ ...
+```
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+  name: jdbc-connection-parameter
+spec:
+  secret:
+    environment: prod
+    name: jdbc-connection-parameter
+ ...
+```
+
+#### Example: If the destination is `Namespace` and secret has to be linked to SA
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: jdbc-connection-parameter
+  annotations:
+      spi.appstudio.redhat.com/secret-linkedto-serviceAccount-name: sa-star
+  labels:
+    spi.appstudio.redhat.com/upload-secret: secret
+    spi.appstudio.redhat.com/target-namespace: ns-9459039345
+type: Opaque
+data:
+ ...
+```
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+  name: jdbc-connection-parameter
+spec:
+  secret:
+    namespace: ns-9459039345
+    name: jdbc-connection-parameter
+    linkedTo:
+    - serviceAccount:
+        reference:
+          name: sa-star
+```
+
+
+### Security and Access Control
+
+* Access to the secret management backend should be restricted to authorized users and components.
```

Comment:
Are there any criteria for who/what is an authorized user or component?

Reply:
Can we document this? Maybe it would be good to pull that table out of the ADR and put it elsewhere in the book so that it can be continued to be updated as role permission/authorization continues to be touched.

Reply:
If I read https://github.com/redhat-appstudio/book/blob/main/ADR/0011-roles-and-permissions.md correctly, maintainers don't have any permission to work with secrets.